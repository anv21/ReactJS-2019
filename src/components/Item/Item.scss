.item_wrapper {
    width: 250px;
    margin: 20px;

    .item_image {
        width: 250px;
        cursor: pointer;
    }

    .item_info {
        display: flex;
        align-items: flex-start;
        justify-content: space-between;
        margin-top: 5px;

        .item_title {
            flex-basis: 70%;
        }
        
        .item_date {
            border: 1px solid gray;
            border-radius: 5px;
            padding: 1px 5px;
        }
    }

    .item_genres {
        color: gray;
        margin-top: 5px;
        font-size: 14px;
    }
    
}
