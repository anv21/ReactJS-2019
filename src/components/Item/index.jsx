import React from 'react';
import PropTypes from 'prop-types';

class Item extends React.Component {
    constructor(props) {
        super(props);
    }

    _onItemClick = () => {
        const {item, onItemClick} = this.props;
        onItemClick(item.id);
    }

    render() {
        const {item} = this.props;
        const {poster_path, title, release_date, genres} = item;

        return (
            <div className="item_wrapper">
                <img className="item_image"
                    src={poster_path}
                    onClick={this._onItemClick}
                />
                <div className="item_info">
                    <span className="item_title">{title}</span>
                    <span className="item_date">{release_date.slice(0, release_date.indexOf('-'))}</span>
                </div>
                <div className="item_genres">{genres.join(' & ')}</div>
            </div>
        )
    }
}

Item.propTypes = {
    item: PropTypes.object,
    poster_path: PropTypes.string,
    title: PropTypes.string,
    release_date: PropTypes.string,
    genres: PropTypes.array
};

Item.defaultProps = {
    item: {},
    poster_path: '',
    title: '',
    release_date: '',
    genres: []
};

export default Item;