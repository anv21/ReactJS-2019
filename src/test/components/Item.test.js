import React from 'react';
import { shallow } from 'enzyme';

import Item from '../../components/Item/index.jsx';

describe('Item', () => {
    const mockGetItem = jest.fn();
    const mockGetItemsByGenre = jest.fn();
    const mockItem = {
        "id": 447365,
        "title": "Guardians of the Galaxy Vol. 3",
        "tagline": "Life finds a way",
        "vote_average": 0,
        "vote_count": 9,
        "release_date": "2020-05-01",
        "poster_path": "https://image.tmdb.org/t/p/w500/ldoY4fTZkGISMidNw60GHoNdgP8.jpg",
        "overview": "The third film based on Marvel's Guardians of the Galaxy.",
        "budget": 0,
        "revenue": 0,
        "genres": [
          "Action",
          "Adventure",
          "Science Fiction"
        ],
        "runtime": 93
    };
    const item = shallow(
        <Item
            item={mockItem}
            getItem={mockGetItem}
            getItemsByGenre={mockGetItemsByGenre}
        />
    );

    it('should be like snapshot', () => {
        expect(item).toMatchSnapshot();
    });

    it('should call getItem function on ItemImage click', () => {
        item.find('.item_image').simulate('click');
        expect(mockGetItem.mock.calls.length).toBe(1);
        expect(mockGetItemsByGenre.mock.calls.length).toBe(1);
    });
});