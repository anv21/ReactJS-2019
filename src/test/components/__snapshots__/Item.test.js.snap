// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item should be like snapshot 1`] = `ShallowWrapper {}`;
